ssl/bio_ssl.c

```diff
@@ -101,14 +102,14 @@ static int ssl_read(BIO *b, char *out, int outl)
 
     BIO_clear_retry_flags(b);
 
-    ret = SSL_read(ssl, out, outl);
+    ret = SSL_read_ex(ssl, out, outl, &readbytes);
 
     switch (SSL_get_error(ssl, ret)) {
     case SSL_ERROR_NONE:
-        if (ret <= 0)
+        if (readbytes == 0)
```

Comment:
I'm getting myself confused with too many PRs in this area at the moment (and the return values are really screwed up). #1763 actually changes things so that reading 0 from the BIO results in -1 being returned from SSL_read(), which would be 0 for SSL_read_ex(). So, on reflection, I think you're right. I'll remove this.